softioc/extension.c

```diff
@@ -77,6 +80,7 @@ static PyObject *get_field_offsets(PyObject *self, PyObject *args)
             dbentry.pflddes->size,
             dbentry.pflddes->field_type);
         PyDict_SetItemString(dict, field_name, ost);
+        Py_XDECREF(ost);
```

Comment:
Is this the right fix?

Reply:
Yes, looks right to me ... except, you can use `Py_DECREF` instead of `Py_XDECREF` -- you _know_ that `ost` is not null at this point.